ConnectionFactory.php

```diff
@@ -45,8 +45,12 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        if (! isset($params['pdo']) && ! isset($params['charset'])) {
+        $connectionOverrideOptions = $params['connection_override_options'] ?? [];
+        unset($params['connection_override_options']);
+
+        if (! isset($params['pdo']) && (! isset($params['charset']) || (! empty($connectionOverrideOptions) && isset($params['url'])))) {
```

Comment:
Harmful use of `empty` detected. Doesn't bring anything, only disadvantages, like silencing error when declaration of variable disappears. I suggest simply
```suggestion
        if (! isset($params['pdo']) && (! isset($params['charset']) || ($connectionOverrideOptions && isset($params['url'])))) {
```